### PR TITLE
Send unit and JS test coverage data to datadog

### PR DESCRIPTION
**Pull request by benpatterson**
Note the metric names:

Reporting metrics to DataDog...
Sending test_eng.coverage.unit.analytics_dashboard ==> 84.3137254902%
Sending test_eng.coverage.javascript.analytics_dashboard ==> 70.2127659574%
Done.

Travis config still needs to be done, but this PR provides the capability. CC @jzoldak @clintonb @dsjen

### CODE REVIEW

File: scripts/report-coverage-to-datadog.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#
+# This script will report various coverage metrics to datadog.
+#
+#
+
+if [ "$TRAVIS_BRANCH" = "master" ] && [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ -n $DATADOG_API_KEY ]
+
+    then
+    echo "Reporting coverage stats to datadog"
+    git clone https://github.com/wedaly/test-metrics
```
Comment (clintonb) [inline scripts/report-coverage-to-datadog.sh:12]: Any reason not to move this to the edX org?

**Comment by clintonb** [comment]
:+1: Minor concern about the location of the repo. Otherwise, this looks good to me.

**Comment by benpatterson** [comment]
Checking on repo location. 

The failed build is unrelated. @dsjen It's a jscs error. See those before?

File: scripts/report-coverage-to-datadog.sh (hunk 1 above)
Comment (benpatterson) [inline scripts/report-coverage-to-datadog.sh:12]: Sounds good to me. I've looped in @wedaly and logged a devops ticket. We can adjust the code whenever that move takes place. The datadog api key will be added separately, and until then, the script will be skipped.

**Comment by benpatterson** [comment]
@clintonb @dsjen that jscs failure comes with the latest version of jscs (version 1.11.0). I'll put in a separate PR to either pin the version or fix the errors.

**Pull request merged**